lib/srv/db/common/auth.go

```diff
@@ -293,27 +293,45 @@ func (a *dbAuth) GetAzureAccessToken(ctx context.Context, sessionCtx *Session) (
 // authority. For onprem we generate a client certificate signed by the host
 // CA used to authenticate.
 func (a *dbAuth) GetTLSConfig(ctx context.Context, sessionCtx *Session) (*tls.Config, error) {
+	dbTLSConfig := sessionCtx.Database.GetTLS()
+
+	// Mode won't be set for older clients. We will default to VerifyFull then - the same as before.
+	switch dbTLSConfig.Mode {
+	case types.DatabaseTLSMode_INSECURE:
+		return getTLSConfigInsecure(), nil
+	case types.DatabaseTLSMode_VERIFY_CA:
+		return a.getTLSConfigVerifyCA(ctx, sessionCtx)
+	default:
+		return a.getTLSConfigVerifyFull(ctx, sessionCtx)
+	}
+}
+
+// getTLSConfigVerifyFull returns tls.Config with full verification enabled ('verify-full' mode).
+// Config also includes database specific adjustment.
+func (a *dbAuth) getTLSConfigVerifyFull(ctx context.Context, sessionCtx *Session) (*tls.Config, error) {
 	tlsConfig := &tls.Config{
 		RootCAs: x509.NewCertPool(),
 	}
-	// Don't set the ServerName when connecting to a MongoDB cluster - in case
-	// of replica set the driver may dial multiple servers and will set
-	// ServerName itself. For Postgres/MySQL we're always connecting to the
-	// server specified in URI so set ServerName ourselves.
+
 	if sessionCtx.Database.GetProtocol() != defaults.ProtocolMongoDB {
+		// Don't set the ServerName when connecting to a MongoDB cluster - in case
+		// of replica set the driver may dial multiple servers and will set
+		// ServerName itself. For Postgres/MySQL we're always connecting to the
+		// server specified in URI so set ServerName ourselves.
 		addr, err := utils.ParseAddr(sessionCtx.Database.GetURI())
 		if err != nil {
 			return nil, trace.Wrap(err)
 		}
 		tlsConfig.ServerName = addr.Host()
 	}
+
 	// Add CA certificate to the trusted pool if it's present, e.g. when
-	// connecting to RDS/Aurora which require AWS CA.
-	if len(sessionCtx.Database.GetCA()) != 0 {
-		if !tlsConfig.RootCAs.AppendCertsFromPEM([]byte(sessionCtx.Database.GetCA())) {
-			return nil, trace.BadParameter("invalid server CA certificate")
-		}
+	// connecting to RDS/Aurora which require AWS CA or when was provided in config file.
+	tlsConfig, err := appendCAToRoot(tlsConfig, sessionCtx)
+	if err != nil {
+		return tlsConfig, trace.Wrap(err)
```

Comment:
Is there a reason to return half-built config?
```suggestion
		return nil, trace.Wrap(err)
```

Reply:
Bad IDE refactor output. Fixed.